### DISCUSSION TITLE
neper make test failed

### DISCUSSION THREAD

**Discussion by hongzzhang** [neperfepx/neper#591] [Q&A] [answered]
Hi, everyone

I have some issues while running ctest and I don't know how to solve them.

Neper version: 4.5.1-2
Gmsh version: 4.4.1 (available with command line gmsh)
Ubuntu version: 20.04 LTS

Failed test list:

        128 - M/cl (Failed)
        129 - M/cl_expr (Failed)
        132 - M/clmin (Failed)
        133 - M/clratio (Failed)
        134 - M/cut (Failed)
        135 - M/dim (Failed)
        136 - M/dim2 (Failed)
        137 - M/dim_expr (Failed)
        138 - M/faset (Failed)
        141 - M/interface (Failed)
        142 - M/interface2 (Failed)
        143 - M/interface3 (Failed)
        144 - M/mesh2dalgo_dela (Failed)
        145 - M/mesh2dalgo_fron (Failed)
        146 - M/mesh2dalgo_mead (Failed)
        147 - M/mesh2dalgo_netg (Failed)
        148 - M/mesh3dalgo_netggmne (Failed)
        149 - M/mesh3dalgo_netggmsh (Failed)
        150 - M/mesh3dalgo_netgnetg (Failed)
        151 - M/meshing (Failed)
        153 - M/meshqualdisexpr (Failed)
        154 - M/meshqualexpr (Failed)
        155 - M/meshqualmin (Failed)
        157 - M/nset1 (Failed)
        158 - M/nset2 (Failed)
        159 - M/nset3 (Failed)
        160 - M/order (Failed)
        161 - M/order_dim2 (Failed)
        166 - M/part_dim2 (Failed)
        167 - M/pl (Failed)
        168 - M/rcl (Failed)
        169 - M/rcl_expr (Failed)
        172 - M/remesh1 (Failed)
        173 - M/remesh2 (Failed)
        174 - M/remesh3 (Failed)
        175 - M/remesh4 (Failed)
        181 - M/statelsetco (Failed)
        187 - M/stateltco (Failed)
        189 - M/tesr_dim2 (Failed)
        192 - M/tess2ddim3 (Failed)
        329 - V/space_pf (Failed)
        330 - V/space_pf_2d (Failed)
        331 - V/space_pf_2d_tesr (Failed)
        332 - V/space_pf_cellcol (Failed)
        333 - V/space_pf_cellcol_tesr (Failed)
        334 - V/space_pf_cellrad (Failed)
        335 - V/space_pf_cellrad_tesr (Failed)
        336 - V/space_pf_cellsymbol_diamond (Failed)
        337 - V/space_pf_cellsymbol_diamond_tesr (Failed)
        338 - V/space_pf_cellsymbol_square (Failed)
        339 - V/space_pf_cellsymbol_square_tesr (Failed)
        340 - V/space_pf_cellsymbol_star (Failed)
        341 - V/space_pf_cellsymbol_star_tesr (Failed)
        342 - V/space_pf_cellsymbol_triangle (Failed)
        343 - V/space_pf_cellsymbol_triangle_tesr (Failed)
        344 - V/space_pf_clustering (Failed)
        345 - V/space_pf_custom_input (Failed)
        346 - V/space_pf_custom_input_ori (Failed)
        347 - V/space_pf_custom_input_ori_des (Failed)
        348 - V/space_pf_custom_input_vect (Failed)
        349 - V/space_pf_custom_input_vect_pfdir (Failed)
        350 - V/space_pf_custom_input_vect_pfdir2 (Failed)
        351 - V/space_pf_density (Failed)
        352 - V/space_pf_density_pfdir (Failed)
        353 - V/space_pf_density_pfdir2 (Failed)
        354 - V/space_pf_densitycolscheme (Failed)
        355 - V/space_pf_densitygridsize (Failed)
        356 - V/space_pf_densitykernel (Failed)
        357 - V/space_pf_densitypoint (Failed)
        358 - V/space_pf_densityscale (Failed)
        359 - V/space_pf_densityscaleticks (Failed)
        360 - V/space_pf_pfdir (Failed)
        361 - V/space_pf_pfdir2 (Failed)
        362 - V/space_pf_pffont (Failed)
        363 - V/space_pf_pfpole (Failed)
        364 - V/space_pf_pfproj (Failed)
        365 - V/space_pf_sym (Failed)
        366 - V/space_pf_tesr (Failed)
        367 - V/space_tree (Failed)
        368 - V/space_tree_res (Failed)
Errors while running CTest

Please help me to solve these. Many thanks.

**Comment by hongzzhang** [comment]
Hi, everyone. When I use the Neper version: 4.3.0, all the tests pass.

**Comment by rquey** [comment, marked as answer]
M tests: There must be something wrong with your Gmsh installation. Go to e.g. `tests/M/cl` and compare `ref.msh` and `test.msh.bak` (if the test did not fail, these files should have been be identical). If `test.msh.bak` does not exist, then it tells you that Neper (or Gmsh) failed - then, simply run the command provided in `test.cmake` to see what the problem is.

V tests: you are missing Asymptote, see [Installing Neper](https://neper.info/doc/introduction.html#installing-neper). You'll also need [Tex Live](https://www.tug.org/texlive) (`texlive-full` on Ubuntu).
